src/main/webapp/app/exercises/programming/shared/code-editor/actions/code-editor-actions.component.ts

```diff
@@ -89,6 +94,14 @@ export class CodeEditorActionsComponent implements OnInit, OnDestroy, OnChanges
             .getBuildingState()
             .pipe(tap((isBuilding: boolean) => (this.isBuilding = isBuilding)))
             .subscribe();
+
+        this.autoSaveInterval = window.setInterval(() => {
+            this.autoSaveTimer++;
+            if (this.autoSaveTimer >= AUTOSAVE_EXERCISE_INTERVAL) {
+                this.autoSaveTimer = 0;
+                this.onSave();
+            }
+        }, AUTOSAVE_CHECK_INTERVAL);
     }
```

Comment:
The reason for autoSaveTimer is to make the timing more consistent, since the client seams to be bad at measuring longer intervals correctly. (Do you know something about this property?)
If the RxJS one is more constant, we could think about using it. Apart from that there should not be any benefit from changing as far as I can tell from this SO post: https://stackoverflow.com/questions/52018881/why-would-i-use-rxjs-interval-or-timer-polling-instead-of-window-setinterval